cmd/syncthing/generate/generate.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2021 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Package generate implements the `syncthing generate` subcommand.
+package generate
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/syncthing/syncthing/cmd/syncthing/cmdutil"
+	"github.com/syncthing/syncthing/lib/config"
+	"github.com/syncthing/syncthing/lib/events"
+	"github.com/syncthing/syncthing/lib/fs"
+	"github.com/syncthing/syncthing/lib/locations"
+	"github.com/syncthing/syncthing/lib/osutil"
+	"github.com/syncthing/syncthing/lib/protocol"
+	"github.com/syncthing/syncthing/lib/syncthing"
+)
+
+type CLI struct {
+	cmdutil.CommonOptions
+	GUIUser     string `placeholder:"STRING" help:"Specify new GUI authentication user name"`
+	GUIPassword string `placeholder:"STRING" help:"Specify new GUI authentication password (use - to read from standard input)"`
```

Comment:
Just noticed while reviewing the docs PR: Maybe we shouldn't allow to specify a pw as an argument at all, only stdin or enter it interactively.

Reply:
I think that would be surprising to most people.  It's easy to work around using `echo foo | ...`, but that has the same security implications (though if someone can read your shell history, they can just as well access your files directly, and I think non-interactive shells don't even save history).  So in an automated use-case, why let the user jump through that extra hoop?

Another UX point: When being asked for a password on the console, I'd expect it to not get echoed while typing.  This feature here is meant mainly for piping (`ssh-askpass | syncthing ...`).  So avoiding the complexities of terminal settings is the goal here.